Handle maven projects without populated dependencies

auke- opened the pull request:
The maven project dependencies are only populated in the maven
project model when the project is in the current reactor.

Don't throw a NullPointerException but ignore dependencies when
they aren't populated.

tbroyer commented:
In which situation would that happen? Could you give example commands that would lead to the NPE? (based on the `e2e`  integration-test reactor project structure, for instance)

Also, that would likely also apply to the `import-sources` and `import-test-sources` mojos. If that's the case, that also means we could have an integration test for that behavior.

auke- commented:
I've created a project with the modular-webapp archetypeArtifactId from your gwt-maven-archetype and added another client app (a new gwt-app maven project and added it as a dependency in the -server webapp).

The error occurs when i run 'mvn gwt:codeserver -pl *-client -am'. It does work when i run 'mvn gwt:codeserver -pl *-server -am' because the new gwt-app is a dependency of the -server webapp and not a dependency of the -client gwt-app.
With the fix the codeserver also works when i run 'mvn gwt:codeserver -pl *-client -am'.

tbroyer commented:
OK, issue reproduced. The bug is actually that we call `project.getCollectedProjects()` when `projects` is empty, and this returns all projects in the reactor, not only in the _reactor build_. Workarounds are either:
* `mvn gwt:codeserver` or `mvn gwt:codeserver -pl *-server -am` or `mvn gwt:codeserver -pl foo-client,bar -am` to have both projects in the reactor build
* pass `-Dgwt.projects=foo-client` to target your project.

tbroyer commented:
Should be fixed by f361ab4dafb39ed503c43e697396d2622636d71d
